cpp/velox/compute/VeloxColumnarToRowConverter.cc

```diff
@@ -40,49 +40,32 @@ arrow::Status VeloxColumnarToRowConverter::init() {
   numRows_ = rv_->size();
   numCols_ = rv_->childrenSize();
 
-  ArrowSchema cSchema{};
-  velox::exportToArrow(rv_, cSchema);
-  ARROW_ASSIGN_OR_RAISE(std::shared_ptr<arrow::Schema> schema, arrow::ImportSchema(&cSchema));
-  if (numCols_ != schema->num_fields()) {
-    return arrow::Status::Invalid("Mismatch: num_cols_ != schema->num_fields()");
-  }
-  schema_ = schema;
-
-  // The input is Arrow batch. We need to resume Velox Vector here.
-  vecs_.clear();
-  resumeVeloxVector();
-
-  // Calculate the initial size
-  nullBitsetWidthInBytes_ = calculateBitSetWidthInBytes(numCols_);
-  int64_t fixedSizePerRow = calculatedFixeSizePerRow(schema_, numCols_);
+  fast_ = std::make_unique<velox::row::UnsafeRowFast>(rv_);
 
   // Initialize the offsets_ , lengths_, buffer_cursor_
   lengths_.clear();
   offsets_.clear();
-  bufferCursor_.clear();
-
-  lengths_.resize(numRows_, fixedSizePerRow);
+  lengths_.resize(numRows_, 0);
   offsets_.resize(numRows_, 0);
-  bufferCursor_.resize(numRows_, nullBitsetWidthInBytes_ + 8 * numCols_);
-
-  // Calculated the lengths_
-  for (int64_t colIdx = 0; colIdx < numCols_; colIdx++) {
-    std::shared_ptr<arrow::Field> field = schema_->field(colIdx);
-    if (arrow::is_binary_like(field->type()->id())) {
-      auto strViews = vecs_[colIdx]->asFlatVector<velox::StringView>()->rawValues();
-      for (int rowIdx = 0; rowIdx < numRows_; rowIdx++) {
-        auto length = strViews[rowIdx].size();
-        int64_t bytes = roundNumberOfBytesToNearestWord(length);
-        lengths_[rowIdx] += bytes;
-      }
+
+  size_t totalMemorySize = 0;
+  if (auto fixedRowSize = velox::row::UnsafeRowFast::fixedRowSize(velox::asRowType(rv_->type()))) {
+    totalMemorySize += fixedRowSize.value() * numRows_;
+
+    for (auto i = 0; i < numRows_; ++i) {
+      lengths_[i] = fixedRowSize.value();
```

Comment:
I don't know why we init it here, not set the length_ and offset_ in function `write`.